realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -587,15 +588,16 @@ public void setList(String fieldName, RealmList<? extends RealmObject> list) {
         long columnIndex = row.getColumnIndex(fieldName);
         LinkView links = row.getLinkList(columnIndex);
         links.clear();
+        Table linkTargetTable = links.getTable().getLinkTarget(columnIndex);
         for (int i = 0; i < list.size(); i++) {
             RealmObject obj = list.get(i);
-            if (!typeValidated) {
-                String elementType = obj.row.getTable().getName();
-                if (!tableName.equals(elementType)) {
-                    throw new IllegalArgumentException(String.format(Locale.ENGLISH,
-                            "Element at index %d is not the proper type. " +
-                            "Was %s expected %s.", i, elementType, tableName));
-                }
+            if (obj.realm != realm) {
+                throw new IllegalArgumentException("Each element in `list` must belong to the same Realm instance.");
```

Comment:
Don't use back quotes- it's not markdown ;-) Just use plain quotation marks (`'`).